### ISSUE TITLE
Glib-alike packages with lib/pkgname/include/pkgnameconfig.h are splitted the wrong way

### ISSUE THREAD

**Issue by abbradar** [NixOS/nixpkgs#14673]
## Issue description

At least `glib` has `lib/glib-2.0/include/glibconfig.h` in its `out` output. It's fine while we use `pkgconfig` but breaks CMake packages which expect it to be in `dev` output. I think it's the right place for it, after all.

I'm slowly searching for those packages and patching them one-by-one but this is a slow manual process. Maybe someone has an idea how to automate it? Also, do we want to do this at all or maybe fix it somehow else?

cc @vcunat

**Comment by abbradar** [comment]
Oh, I've found an issue -- those headers are used by some internal Glib stuff too. My new strategy is just symlinking `$out/lib/pkgname` to `$dev/lib/pkgname`.

**Comment by abbradar** [comment]
Phew, it seems I was wrong -- there are some packages which do this but I've found only `dbus`, `gtk2` and `glib` which are both splitted and have include files in `/lib`.

**Comment by vcunat** [comment]
From this it's not clear to me what you've found out. I *had* been puzzled why the packages insist on putting these headers into `--libdir`, but these headers seemed small enough for me not to try to override that default.

**Comment by abbradar** [comment]
Sorry, it was late in the night and I should have pointed out more ^^. The story goes like this:

1) I've noticed problem with finding these headers by CMake and opened this.
2) Thought it would be good to move them to `dev`, but feared that there might be many of packages like this.
3) Moved it for `glib` but encountered a problem later when some `glibc` utility was not able to find this header.
4) Moved to just symlinking it to `dev` to have CMake find them automatically.
5) After some time, understood that (luckily -- that's what I meant in the last message) there are not many packages that we split and that move headers there.
6) After yet more time understood that my solution fixes nothing -- CMake still has to be pointed at these headers. So I've commited https://github.com/NixOS/nixpkgs/commit/755e4633c7e6778f9883f005930a5af471945151 and closed this.
